v2/component/launcher.go

```diff
@@ -228,11 +230,17 @@ func NewLauncher(runtimeInfo string, options *LauncherOptions) (*Launcher, error
 		return nil, err
 	}
 
+	cacheClient, err := cacheutils.NewClient()
+	if err != nil {
+		return nil, err
+	}
+
 	return &Launcher{
 		options:                 options,
 		placeholderReplacements: pr,
 		runtimeInfo:             rt,
 		metadataClient:          metadataClient,
+		cacheClient:             cacheClient,
```

Comment:
for discussion, shall we add a launcher flag for enabling cache?
we can keep the flag disabled during implementation and turn it on in the last PR that finishes all the implementation (or depending on sdk config)

this avoids unfinished caching logic to have any effect on presubmit tests etc

Reply:
I was planning to finish all cache logic in this PR. We will need a launcher flag for specifying enabling cache or not. It will be implemented in https://github.com/kubeflow/pipelines/issues/5818

Reply:
Makes sense to me!